services/galley/src/Galley/API/Teams/Features.hs

```diff
@@ -635,15 +634,14 @@ getGuestLinkInternal = \case
   Left _ -> getCfgDefault
   Right tid -> do
     cfgDefault <- getCfgDefault
-    let defLockStatus = Public.tfwoapsLockStatus cfgDefault
-    maybeFeatureStatus <- TeamFeatures.getFeatureStatusNoConfig @'Public.TeamFeatureGuestLinks tid
-    pure $ case (defLockStatus, maybeFeatureStatus) of
+    (mbFeatureStatus, fromMaybe (Public.tfwoapsLockStatus cfgDefault) -> lockStatus) <- TeamFeatures.getFeatureStatusNoConfigAndLockStatus @'Public.TeamFeatureGuestLinks tid
+    pure $ case (lockStatus, mbFeatureStatus) of
       (Public.Unlocked, Just featureStatus) ->
         Public.TeamFeatureStatusNoConfigAndLockStatus
           (Public.tfwoStatus featureStatus)
-          Public.Unlocked
-      (Public.Unlocked, Nothing) -> cfgDefault {Public.tfwoapsLockStatus = Public.Unlocked}
-      (Public.Locked, _) -> cfgDefault {Public.tfwoapsLockStatus = Public.Locked}
+          lockStatus
+      (Public.Unlocked, Nothing) -> cfgDefault {Public.tfwoapsLockStatus = lockStatus}
+      (Public.Locked, _) -> cfgDefault {Public.tfwoapsLockStatus = lockStatus}
```

Comment:
[nit-pick] not sure i like this change.  it's less direct.  (then again it's more resilient to change in the pattern, possibly.)